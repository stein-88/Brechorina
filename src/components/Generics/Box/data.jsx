const schema = [
  'section',
  'container',
  'row',
  'col',
  'span',
];

export default schema;
