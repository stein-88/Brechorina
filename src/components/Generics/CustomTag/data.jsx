import StandardButton from '../StandardButton'
import GenClickBox from '../GenClickBox'
import GenericInput from '../GenericInput'

const validTags = [
    'p',
    'a',
    'ul',
    'li',
    'span',
    'img',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'object',
    'strong',
    'small',
    'input',
    'label',
    'em',
    'hr',
    'br',
    'form',
    'button',
    'iframe',
    'video',
    'textarea',
    'i',
    'sup',
    'picture',
    'source',
    'div',
    'section',
    'comp',
]

export default {
    validTags,
    StandardButton,
    GenClickBox,
    GenericInput
}
