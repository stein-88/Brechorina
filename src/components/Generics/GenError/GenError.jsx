import React from 'react'
import { helperCloser } from '@utils'
import StandardButton from '../StandardButton/StandardButton'
import GenErrorSchema from './data'
import s from './GenError.scss'

const GenError = ({
    message, noButton, customClass,
}) => {
    if (!message || !GenErrorSchema) return null
    const { someWrong, detail, closeButton } = GenErrorSchema
    return (
        <section className={`w-100 ${customClass || ''}`}>
            <div className="container">
                <div className="row">
                    <div className="col-12">
                        <p className={`${s.bold} h1 pb-4`}>{someWrong} <i className={`fas ml-3 ${s.txtError} align-middle fa-exclamation-circle`} /></p>
                    </div>
                </div>
                <div className="row pb-4">
                    <div className="col-12 pb-4">
                        <p className="pb-3 mb-0 h5">{detail}:</p>
                        <p className="pb-3 h5"><span className={s.txtError}>{message}</span></p>
                    </div>
                    {!noButton
                        && (
                            <div className={`col-12 text-right ${s.globeLight}`}>
                                <StandardButton
                                    darkBlue
                                    onClick={() => helperCloser()}
                                    text={closeButton}
                                    altern
                                />
                            </div>
                        )}
                </div>
            </div>
        </section>
    )
}

export default GenError
