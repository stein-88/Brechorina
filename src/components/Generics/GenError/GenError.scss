.bold {
    font-family: MyRalewayBold, sans-serif;
}

.txtError {
    color: red;
}

.globeLight {
    font-family: MyRalewayLight, sans-serif;
}