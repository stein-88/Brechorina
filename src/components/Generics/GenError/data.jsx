const GenErrorSchema = {
    someWrong: 'Something went wrong!',
    detail: 'Details',
    closeButton: 'Close',
}

export default GenErrorSchema
