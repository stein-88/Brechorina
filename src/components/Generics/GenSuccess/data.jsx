const genSuccesSchema = {
    succTitle: 'Success!',
    closeButton: 'Close',
}

export default genSuccesSchema
