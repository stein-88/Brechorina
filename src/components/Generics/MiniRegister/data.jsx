import s from './MiniRegister.scss'

const joinForm = [
    {
        data: [
            {
                type: 'text',
                value: 'joinFirstName',
                placeHolder: 'First Name',
                customClass: `py-2 ${s.inpStyle}`,
                required: true,
            },
            {
                type: 'text',
                value: 'joinLastName',
                placeHolder: 'Last Name',
                customClass: `py-2 ${s.inpStyle}`,
                required: true,
            },
        ]
    },
    {
        data: [
            {
                type: 'text',
                value: 'joinEmail',
                placeHolder: 'Email',
                customClass: `py-2 ${s.inpStyle}`,
            },
            {
                type: 'text',
                value: 'joinContactNum',
                placeHolder: 'Telephone Number',
                customClass: `py-2 ${s.inpStyle}`,
            },
        ]
    },
    {
        fullCol: true,
        data: [
            {
                type: 'select',
                value: 'joinMemeberType',
                placeHolder: 'Member Type',
                customClass: `${s.inpStyle} mb-0 py-2`,
                options: [
                    'Individual',
                    'Charity',
                    'Business',
                ],
            },
        ]
    }
]

export default joinForm
