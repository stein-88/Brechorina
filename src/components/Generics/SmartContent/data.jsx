import GCCarousel from '@generics/GCCarousel'

const mapComp = {
    GCCarousel,
}

const schema = {
    permComp: ['box', ...Object.keys(mapComp)],
    mapComp,
}

export default schema
