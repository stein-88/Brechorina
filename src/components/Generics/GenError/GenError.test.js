import React from 'react'
import renderer from 'react-test-renderer'
import GenError from './GenError'
import * as schemaMocked from './data'

const localSchema = { ...schemaMocked, default: null }

describe('Test on <GenError /> component', () => {
    test('Should render without props', () => {
        const componentNoprops = renderer.create(<GenError />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toBeNull()
    })
    test('Should render with full props', () => {
        const componentNoprops = renderer.create(<GenError message="Message" noButton correctContainer />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
    test('Should not render without the schema', () => {
        localSchema.default = null
        const componentNoprops = renderer.create(<GenError />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toBeNull()
    })
})
