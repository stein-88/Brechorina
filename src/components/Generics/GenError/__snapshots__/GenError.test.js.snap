// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GenError /> component Should render with full props 1`] = `
<section
  className="w-100 "
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <p
          className="bold h1 pb-4"
        >
          Something went wrong!
           
          <i
            className="fas ml-3 txtError align-middle fa-exclamation-circle"
          />
        </p>
      </div>
    </div>
    <div
      className="row pb-4"
    >
      <div
        className="col-12 pb-4"
      >
        <p
          className="pb-3 mb-0 h5"
        >
          Details
          :
        </p>
        <p
          className="pb-3 h5"
        >
          <span
            className="txtError"
          >
            Message
          </span>
        </p>
      </div>
    </div>
  </div>
</section>
`;
